includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1045,14 +1053,12 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 				}
 
 				// For $0 orders, we need to save the payment method using a setup intent.
-				$intent = $this->payments_api_client->create_and_confirm_setup_intent(
-					$payment_information->get_payment_method(),
-					$customer_id,
-					false,
-					$this->is_platform_payment_method( $payment_information->is_using_saved_payment_method() ),
-					$save_user_in_platform_checkout,
-					$metadata
-				);
+				$create_and_confirm_intent_request = Create_And_Confirm_Setup_Intention::create();
+				$create_and_confirm_intent_request->set_customer( $customer_id );
+				$create_and_confirm_intent_request->set_payment_method( $payment_information->get_payment_method() );
+				$create_and_confirm_intent_request->set_metadata( $metadata );
+				$intent = $create_and_confirm_intent_request->send( 'wcpay_create_and_confirm_setup_intention_request', $payment_information, false, $save_user_in_platform_checkout );
```

Comment:
This is really a nitpick, but could we just use `$request` here? Having half of the line occupied by the variable name feels very wasteful.